docs/development/design/automation-rules.rst

```diff
@@ -0,0 +1,101 @@
+Automation Rules
+================
+
+Automation rules allow users to apply some actions over new or deleted versions.
+See :doc:`/automation-rules` for more details of the current features.
+
+This document explore some ideas to expand automation rules to automate more tasks.
+
+Current state and limitations
+-----------------------------
+
+- Matches are done using regular expressions.
+  But the models were designed so other type of matches can be implemented
+  (like comparing the version with the highest semver version).
+- Automation rules are meant to automate existing actions,
+  so an action shouldn't be able to be triggered only inside an automation rule.
+  This is so users aren't blocked if a rule didn't match a version,
+  or if they need to execute the action over existing versions.
+- Automation rules are currently matched against new versions,
+  but they can also be executed over deleted versions (https://github.com/readthedocs/readthedocs.org/pull/7644).
+
+Improvements
+------------
+
+Keep track of the ``n`` latest matches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We do this with webhook exchanges (keeping the latest 10)
+https://github.com/readthedocs/readthedocs.org/blob/f5b76e87a3b970d42228f5972ae9a50364c3c76c/readthedocs/integrations/models.py#L99-L111.
+See https://github.com/readthedocs/readthedocs.org/issues/6393.
+
+Live feedback of versions that match a rule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When creating/editing a rule, it would be useful to know what versions are going to match.
+We could list the current versions that match and/or allow the user to enter a custom input.
+A new enpoint in api v2/v3 should be added.
+
+Easy way to invert a match
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, users need to write a negative look ahead,
+which isn't really user-friendly or easy to remember/write.
+Having an explicit option to invert a match would be more useful,
+see https://github.com/readthedocs/readthedocs.org/issues/6354.
+
+New features
+------------
+
+Change the privacy level of external versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Read the Docs for business all versions are private by default.
+Some users may want to change the privacy level of external versions.
+See https://github.com/readthedocs/readthedocs-corporate/issues/1063.
+
+If automation rules are going to be used, users will need to be able to edit those versions.
+This means having a place to list all external versions and allow to edit some properties,
+privacy level is the only option that makes sense maybe also allow to de-activate it so docs aren't accessible.
```

Comment:
I'm not sure if it makes sense to "de-activate PR via automation rule". I think users can already disable this feature from the project admin. Is this different than the current disable option there?

Reply:
Here the deactivating the version would delete the files from storage. Currently, we don't delete them. I think we may want to have this in .com, so the content isn't flying around

Reply:
I don't follow here. When a PR is merged, we do delete everything for that external version, including builds.

Can you elaborate what would be a good use case for the feature of "de-activating external versions via automated rules"?

Reply:
> I don't follow here. When a PR is merged, we do delete everything for that external version, including builds.

We only delete the build logs, the documentation is kept in storage. But we went with another solution for now https://github.com/readthedocs/readthedocs.org/pull/7678